src/wrappers/bunnybus.js

```diff
@@ -0,0 +1,144 @@
+    let originalHandlerSyncErr;
+    let runnerResult;
+    try {
+        // Initialize tracer and runner.
+        tracer.restart();
+        const { slsEvent: amqpEvent, startTime: amqpStartTime } =
+        eventInterface.initializeEvent(
+            'rabbitmq',
+            args[1].headers.routeKey,
+            'consume',
+            'trigger'
+        );
+
+        const metadata = {
+            host: config.hostname,
+            vhost: config.vhost,
+            'messaging.message_payload_size_bytes': JSON.stringify(args[0]).length,
+        };
+        if (args[1].headers[EPSAGON_HEADER]) {
+            metadata[EPSAGON_HEADER] = args[1].headers[EPSAGON_HEADER].toString();
+        }
+
+        tracer.addEvent(amqpEvent);
+        eventInterface.finalizeEvent(amqpEvent, amqpStartTime, null, metadata, {
+            headers: args[1].headers,
+            message: args[0],
+        });
+
+        const { label, setError } = tracer;
+        // eslint-disable-next-line no-param-reassign
+        args.push({
+            label,
+            setError,
+        });
+        const runnerName = callback && callback.name ? callback.name : `${topic}-consumer`;
+        const { slsEvent: nodeEvent, startTime: nodeStartTime } = eventInterface.initializeEvent(
+            'node_function', runnerName, 'execute', 'runner'
+        );
+
+        try {
+            runnerResult = callback(...args);
+        } catch (err) {
+            originalHandlerSyncErr = err;
+        }
+
+        // Handle and finalize async user function.
+        if (utils.isPromise(runnerResult)) {
+            let originalHandlerAsyncError;
+            runnerResult = runnerResult.catch((err) => {
+                originalHandlerAsyncError = err;
+                throw err;
+            }).finally(() => {
+                eventInterface.finalizeEvent(nodeEvent, nodeStartTime, originalHandlerAsyncError);
+                tracer.sendTrace(() => {});
+            });
+        } else {
+            // Finalize sync user function.
+            eventInterface.finalizeEvent(nodeEvent, nodeStartTime, originalHandlerSyncErr);
+            tracer.sendTrace(() => {});
+        }
+        tracer.addRunner(nodeEvent, runnerResult);
+    } catch (err) {
+        tracer.addException(err);
+    }
+    // Throwing error in case of sync user function.
+    if (originalHandlerSyncErr) {
+        throw originalHandlerSyncErr;
+    }
+    return runnerResult;
+}
+
+/**
+ * Wraps the BunnyBus callback and channel consumer creation to wrap the run function
+ * @param {Function} wrappedFunction The BunnyBus subscribe function
+ * @returns {Function} The wrapped function
+ */
+function bunnybusConsumerWrapper(wrappedFunction) {
+    traceContext.init();
+    tracer.getTrace = traceContext.get;
+    return function internalBunnybusConsumerWrapper(queue, handlers, options) {
+        if (options.meta) {
+            // Enabling tracing only if meta is enabled.
```

Comment:
Is it possible to enable tracking when `options.meta` is false?  We don't ever turn on `options.meta` in our codebase atm because we haven't encountered a need yet to retrieve info from the header.